### PR TITLE
Only search relative paths when @-mentioning files in chat

### PR DESCRIPTION
**Pull request by DanTup**
[This depends on a function added in https://github.com/sourcegraph/cody/pull/2215 so this will fail bots until that lands and I rebase/merge]

Fixes https://github.com/sourcegraph/cody/issues/2208

## Test plan

- Run `pnpm test:e2e at-file` to run the updated test, or
- Open chat, and mention a file using `@` but type part of the file path outside of the folder opened in VS Code (eg. if you have `C:\Source\cody\ open, type `source`)
- Ensure that the results don't list every file because `source` exists in their full fspaths

### CODE REVIEW

File: vscode/test/e2e/at-file-context-selecting.test.ts [hunk 1]
```diff
@@ -22,6 +22,15 @@ test('@-file empty state', async ({ page, sidebar }) => {
     await chatInput.fill('@definitelydoesntexist')
     await expect(chatPanelFrame.getByRole('heading', { name: 'No matching files found' })).toBeVisible()
 
+    // Clear the input so the next test doesn't detect the same text already visible from the previous
+    // check (otherwise the test can pass even without the filter working).
+    await chatInput.clear()
+
+    // We should only match the relative visible path, not parts of the full path outside of the workspace.
+    // Eg. searching for "source" should not find all files if the project is inside `C:\Source`.
+    await chatInput.fill('@fixtures') // fixture is in the test project folder name, but in the relative paths.
```
Comment (dominiccooney) [inline vscode/test/e2e/at-file-context-selecting.test.ts:31]: Once #2235 lands, you can make this more robust by adding `workspaceDirectory` alongside `page, sidebar` and get the path to the workspace directory. That will let you assert that fixtures really *is* in that path and remains that way.

**Review by dominiccooney — APPROVED** [review]
Looks good!

File: vscode/test/e2e/at-file-context-selecting.test.ts (hunk 1 above)
Comment (DanTup) [reply vscode/test/e2e/at-file-context-selecting.test.ts:36]: Oh yeah, good point. I've added a TODO about that. If you're planning to land that soon I can wait and fix it up here, or if you think I should land this as-is and fix later, let me know.

I've rebased this on main after merging #2208 so I think this should all be green now.

**Comment by DanTup** [comment]
Hmm, there's something odd going on now. With this change, `.vscode\settings.json\` shows up in the search results for `@vgo` (it has no `g`!), which results in the test selecting the wrong item - but that also results in `and @vg` being duplicated on the end:

https://github.com/sourcegraph/cody/assets/1078012/3c833a18-cfa1-4776-867b-59fbad9610bf

This is reliably reproducible for me, and feels like two bugs (non-matched items in the results, and the duplicated text). I might not get chance to look into this before Friday now, so if this is important and anyone else has time, feel free to take a look!

**Comment by DanTup** [comment]
> it has no `g`!

lol, it very much as a `g` 😄 I'll update the test and file another bug about the duplication (which seems to occur only when you select a file that's already listed).

**Comment by DanTup** [comment]
I've filed https://github.com/sourcegraph/cody/issues/2243 about the bug.

@dominiccooney heading off, but if you're happy with this as-is and the last bots go green feel free to merge :)

**Pull request merged**